service/resources/arcgis.py

```diff
@@ -0,0 +1,103 @@
+"""ArcGIS module"""
+import os
+import math
+import json
+import requests
+import falcon
+import jsend
+import usaddress
+
+class Arcgis():
+    """Welcome class"""
+    def on_get(self, req, resp, name):
+        """on get request
+        return Arcgis message
+        """
+        if name in dir(self):
+            return getattr(self, name)(req, resp)
+        msg = {'message': 'Welcome'}
+        resp.body = json.dumps(jsend.success(msg))
+        resp.status = falcon.HTTP_200
+
+    def parcels(self, req, resp):
+        """ get parcels """
+
+        missing_env = self.has_missing_env(['PLN_ARCGIS_PARCEL'])
+        if missing_env:
+            resp.status = falcon.HTTP_404
+            msg_error = jsend.error(
+                'Missing or invalid environment variable(s): '+", ".join(missing_env))
+            resp.body = json.dumps(msg_error)
+            return
+
+        response = {'message': 'parcels'}
+        if 'address' in req.params:
+            addr = usaddress.tag(req.params['address'])
+            options = {'returnGeometry':'false', 'returnSuggestions':False}
+            if 'returnSuggestions' in req.params and req.params['returnSuggestions'] == 'true':
+                options['returnSuggestions'] = True
+            if 'returnGeometry' in req.params:
+                options['returnGeometry'] = req.params['returnGeometry']
+            parcels = self.pln_parcel(addr, options)
+            response = {'parcels': parcels}
+        resp.body = json.dumps(jsend.success(response))
+        resp.status = falcon.HTTP_200
+
+    def pln_parcel(self, addr, options=None):
+        """ get parcels from Planning ArcGIS """
+        parcels = {}
+        url = os.environ.get('PLN_ARCGIS_PARCEL')+'query'
```

Comment:
This assumes that the env var ends with "/".  What happens when it doesn't?

Reply:
commit bf10e31362024045178aaef0ac51197f7a4d3fa1